Sources/Plasma/NucleusLib/pnInputCore/plKeyMap.h

```diff
@@ -214,24 +217,30 @@ class plKeyMap : public plInputMap
         // Clears ALL bindings
         void    ClearAll();
 
-        static const char* GetStringCtrl();
-        static const char* GetStringShift();
-        static const char* GetStringUnmapped();
+        static ST::string GetStringCtrl();
+        static ST::string GetStringShift();
+        static ST::string GetStringUnmapped();
 
 
         size_t              GetNumBindings() const { return fBindings.size(); }
         const plKeyBinding  &GetBinding(size_t i) const { return *fBindings[i]; }
         void                HandleAutoDualBinding( plKeyDef key1, plKeyDef key2 );
 
-        static const char* ConvertVKeyToChar( uint32_t vk );
-        static plKeyDef ConvertCharToVKey( const char *c );
+        static const std::map<uint32_t, ST::string>& GetKeyConversion();
+
+        static ST::string ConvertVKeyToChar(uint32_t vk);
+        static plKeyDef ConvertCharToVKey(const ST::string& c);
+
+        static ST::string KeyComboToString(const plKeyCombo &combo);
+        static plKeyCombo StringToKeyCombo(const ST::string& keyStr);
 
-        static Win32keyConvert  fKeyConversionEnglish[];
-        static Win32keyConvert  fKeyConversionFrench[];
-        static Win32keyConvert  fKeyConversionGerman[];
-        //static Win32keyConvert  fKeyConversionSpanish[];
-        //static Win32keyConvert  fKeyConversionItalian[];
+        static const std::map<uint32_t, ST::string> fKeyConversionEnglish;
+        static const std::map<uint32_t, ST::string> fKeyConversionFrench;
+        static const std::map<uint32_t, ST::string> fKeyConversionGerman;
+        static const std::map<uint32_t, ST::string> fKeyConversionSpanish;
+        static const std::map<uint32_t, ST::string> fKeyConversionItalian;
 
+        static const std::map<plLocalization::Language, const std::map<uint32_t, ST::string>*> fKeyConversions;
```

Comment:
Are these ever actually used directly from outside of the class, or could they be declared file-static in the C++ file rather than class-static in the header file?

That would eliminate the need to bring the PubUtilLib header into this one